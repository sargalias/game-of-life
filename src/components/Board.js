import React from 'react';
import Row from './Row';

const Board = (props) => (
  <div className="board">
    {props.boardData.map((rowData, index) => (
      <Row key={index} rowId={index} rowData={rowData} />
    ))}
  </div>
);

export default Board;
