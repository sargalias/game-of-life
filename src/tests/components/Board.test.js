import React from 'react';
import { shallow } from 'enzyme';
import Board from '../../components/Board';

test('board should render correctly with empty boardData', () => {
  const boardData = [];
  const wrapper = shallow(<Board boardData={boardData} onCellClick={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('board should render correctly with boardData', () => {
  const boardData = [
    [0, 0, 0],
    [1, 0, 1],
    [0, 1, 0]
  ];
  const wrapper = shallow(<Board boardData={boardData} onCellClick={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('board should initialise with a color array in state if props.color=true', () => {
  const boardData = [
    [0, 0, 0],
    [1, 0, 1],
    [0, 1, 0]
  ];
  const wrapper = shallow(<Board color={true} boardData={boardData} />);
  expect(wrapper.state('colors')).toBeTruthy();
});

test('board should not initialise with a color array in state if props.color=false', () => {
  const boardData = [
    [0, 0, 0],
    [1, 0, 1],
    [0, 1, 0]
  ];
  const wrapper = shallow(<Board boardData={boardData} />);
  expect(wrapper.state('colors')).toBeFalsy();
});
