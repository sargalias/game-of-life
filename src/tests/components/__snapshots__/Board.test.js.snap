// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`board should render correctly with boardData 1`] = `
<div
  className="board"
>
  <Row
    key="0"
    rowData={
      Array [
        0,
        0,
        0,
      ]
    }
    rowId={0}
  />
  <Row
    key="1"
    rowData={
      Array [
        1,
        0,
        1,
      ]
    }
    rowId={1}
  />
  <Row
    key="2"
    rowData={
      Array [
        0,
        1,
        0,
      ]
    }
    rowId={2}
  />
</div>
`;

exports[`board should render correctly with empty boardData 1`] = `
<div
  className="board"
/>
`;
